From f9cd18b3e2be679710f66bd02a27e0bfe3f724aa Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Mon, 29 Oct 2018 17:08:09 +0800
Subject: [PATCH 4952/5242] MLK-20121 drm/imx: dpu: plane: Improve readability
 for helper source_to_id()

commit  644a119a490a741a3240d61a737204ab9e1b97f5 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch contains code-change-only to improve readability
for helper source_to_id().  Less indents are used.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dpu/dpu-plane.h |   23 +++++++++++------------
 1 file changed, 11 insertions(+), 12 deletions(-)

diff --git a/drivers/gpu/drm/imx/dpu/dpu-plane.h b/drivers/gpu/drm/imx/dpu/dpu-plane.h
index aba805a..1cc8822 100644
--- a/drivers/gpu/drm/imx/dpu/dpu-plane.h
+++ b/drivers/gpu/drm/imx/dpu/dpu-plane.h
@@ -127,21 +127,20 @@ static inline int source_to_id(lb_sec_sel_t source)
 	int type = source_to_type(source);
 
 	for (i = 0; i < ARRAY_SIZE(sources); i++) {
-		if (source == sources[i]) {
-			if (type == DPU_PLANE_SRC_FD ||
-			    type == DPU_PLANE_SRC_FW) {
-				while (offset < ARRAY_SIZE(sources)) {
-					if (source_to_type(sources[offset]) ==
-					    type)
-						break;
-					offset++;
-				}
-
-				i -= offset;
-			}
+		if (source != sources[i])
+			continue;
 
+		/* FetchLayer */
+		if (type == DPU_PLANE_SRC_FL)
 			return i;
+
+		/* FetchWarp or FetchDecode */
+		while (offset < ARRAY_SIZE(sources)) {
+			if (source_to_type(sources[offset]) == type)
+				break;
+			offset++;
 		}
+		return i - offset;
 	}
 
 	WARN_ON(1);
-- 
1.7.9.5

